kernel/kernel-api/src/main/java/io/delta/kernel/client/FileSystemClient.java

```diff
@@ -56,17 +56,15 @@ public interface FileSystemClient {
      */
     String resolvePath(String path) throws IOException;
 
-    // TODO: solidify input type; need some combination of path, offset, size
-
     /**
-     * Read data specified by the start and end offset from the file. It is the responsibility
-     * of the caller close each returned stream.
+     * Return an iterator of byte streams one for each read request in {@code readRequests}.
+     * It is the responsibility of the caller to close each returned stream.
      *
-     * @param iter Iterator for tuples (file path, range (start offset, end offset)
-     * @return Data for each range requested as one {@link ByteArrayInputStream}.
+     * @param readRequests Iterator of read requests
+     * @return Data for each request as one {@link ByteArrayInputStream}.
```

Comment:
Mention the order of the returned stream is the same as the given file read requests.